homeassistant/components/opengarage/__init__.py

```diff
@@ -36,3 +48,32 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+class OpenGarageDataUpdateCoordinator(update_coordinator.DataUpdateCoordinator):
+    """Class to manage fetching Opengarage data."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        *,
+        open_garage_connection: opengarage.OpenGarage,
+    ) -> None:
+        """Initialize global Opengarage data updater."""
+        self.open_garage_connection = open_garage_connection
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=30),
+        )
+
+    async def _async_update_data(self) -> None:
```

Comment:
This method should not be overridden. Instead specify an `async_update` function and assign it to the `update_method` in `super().__init__`. 
An example to follow: https://github.com/home-assistant/core/blob/dev/homeassistant/components/speedtestdotnet/__init__.py

Reply:
Could you explain why that is preferable?
It seems that several integrations use the current approach

Reply:
I recommend to refer to the developers documentation on [fetching data](https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities) using the DataUpdateCoordinator.

Reply:
That is only covers the case where DataUpdateCoordinator is used directly, and not a custom coordinator inheriting DataUpdateCoordinator

Reply:
Actually, I think the current implementation in speedtestdotnet might be wrong.

I do not think it is a good idea to override the async_update method.
https://github.com/home-assistant/core/blob/dev/homeassistant/components/speedtestdotnet/__init__.py#L127